sdk/core/Azure.Core/src/ResponseClassifier.cs

```diff
@@ -57,5 +60,23 @@ public virtual bool IsErrorResponse(HttpMessage message)
             var statusKind = message.Response.Status / 100;
             return statusKind == 4 || statusKind == 5;
         }
+
+        internal static (string Message, string? ErrorCode) GetErrorDetails(Response response)
```

Comment:
Shoud we inline this logic into ResponseFailedException for now? The ResponseFailedException is the only consumer of it and the method is static.